Spec/Auth.swift

```diff
@@ -1963,40 +1968,43 @@ class Auth : QuickSpec {
                         expect(tokenDetails.clientId).to(equal(options.clientId))
                         expect(tokenDetails.issued!.dateByAddingTimeInterval(defaultTtl)).to(beCloseTo(tokenDetails.expires))
                         expect(tokenDetails.capability).to(equal(defaultCapability))
-                        done()
-                    }
-                }
-
-                tokenParams.ttl = ExpectedTokenParams.ttl
-                tokenParams.capability = ExpectedTokenParams.capability
-                tokenParams.clientId = nil
-
-                let authOptions = ARTAuthOptions()
-                authOptions.force = true
-                authOptions.queryTime = true
-
-                var serverDate = NSDate()
-                waitUntil(timeout: testTimeout) { done in
-                    rest.time { date, error in
-                        expect(error).to(beNil())
-                        guard let date = date else {
-                            XCTFail("No server time"); done(); return
+                        
+                        let authOptions = ARTAuthOptions()
+                        authOptions.authCallback = nil
+                        
+                        rest.auth.authorise(nil, options: authOptions) { tokenDetails2, error in
+                            expect(error).to(beNil())
+                            guard let tokenDetails2 = tokenDetails2 else {
+                                XCTFail("TokenDetails2 is nil"); done(); return
+                            }
+                            expect(rest.auth.options.authCallback).to(beNil())
```

Comment:
It would be more useful here to check that a documented API is overriden i.e. `auth.options` is undocumented, so instead why not check that the previous callback has not been used perhaps, or in the callback set a specific capability and ensure that capability has now not been used thus proving the callback has not been called.